storage/internal/benchmarks/w1r3.go

```diff
@@ -68,26 +68,37 @@ type w1r3 struct {
 }
 
 func (r *w1r3) setup() error {
+	// Select API first as it will be the same for all writes/reads
+	api := opts.api
+	if api == mixedAPIs {
+		if randomBool() {
+			api = xmlAPI
+		} else {
+			api = grpcAPI
+		}
+	}
+
+	// Select object size
 	objectSize := opts.objectSize
 	if objectSize == 0 {
 		objectSize = randomInt64(opts.minObjectSize, opts.maxObjectSize)
 	}
 	r.writeResult = &benchmarkResult{objectSize: objectSize}
-	r.readResults = []*benchmarkResult{{}, {}, {}}
+	r.readResults = []*benchmarkResult{{objectSize: objectSize}, {objectSize: objectSize}, {objectSize: objectSize}}
```

Comment:
Does this mean we haven't been associating an object size with subsequent reads before this PR?

Reply:
We have been, I just moved the assignment to here so it'd be together.